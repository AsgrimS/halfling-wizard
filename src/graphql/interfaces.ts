import {
  Class,
  Subclass,
  Proficiency,
  ClassProficiency_Choices,
  AbilityScoreSkills,
  ClassSpellcasting,
  ClassStarting_Equipment,
} from "./schema";

export interface HeroClassesData {
  classes: Class[];
}

export interface HeroClassData {
  class: Class;
}

export interface SubclassesData {
  subclasses: Subclass[];
}

export interface ProficienciesData {
  proficiencies: Proficiency[];
}

export interface ProficienciesChoicesData {
  proficienciesChoices: ClassProficiency_Choices[];
}
export interface SavingThrowsData {
  saving_throws: AbilityScoreSkills[];
}

export interface ClassSpellcastingData {
  spellcasting: ClassSpellcasting;
}

export interface StartingEquipmentData {
  startingEquipments: ClassStarting_Equipment[];
}
