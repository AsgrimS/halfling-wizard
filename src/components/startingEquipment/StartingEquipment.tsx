import React from "react";
import { v4 as uuidv4 } from "uuid";

import { StartingEquipmentData } from "../../graphql/interfaces";

const StartingEquipment: React.FC<StartingEquipmentData> = ({
  startingEquipments,
}) => {
  return (
    <div data-testid="starting-equipment">
      <span>Starting Equipment</span>
      <ul>
        {startingEquipments.map(({ equipment, quantity }) => (
          <li key={uuidv4()}>
            {equipment!.name}: {quantity}
          </li>
        ))}
      </ul>
    </div>
  );
};

export default StartingEquipment;
