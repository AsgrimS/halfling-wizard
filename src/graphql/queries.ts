import { gql } from "@apollo/client";

export const HERO_CLASS = gql`
  query GetHeroesClasses($classIndex: String!) {
    class(filter: { name: $classIndex }) {
      hit_die
      index
      name
      proficiencies {
        index
        name
      }
      saving_throws {
        index
        name
      }
      proficiency_choices {
        choose
        from {
          name
          index
        }
      }
      spellcasting {
        level
        spellcasting_ability {
          index
          name
        }
      }
      starting_equipment {
        equipment {
          name
          index
        }
        quantity
      }
      starting_equipment_options {
        choose
        from {
          equipment {
            name
            index
          }
          quantity
        }
      }
      subclasses {
        index
        name
      }
    }
  }
`;

export const HEROES_CLASSES = gql`
  query GetHeroesClasses {
    classes {
      name
      index
    }
  }
`;
