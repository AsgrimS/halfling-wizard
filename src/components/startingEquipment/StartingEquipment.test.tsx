import React from "react";
import { render, screen, cleanup } from "@testing-library/react";

import StartingEquipment from "./StartingEquipment";
import { ClassStarting_Equipment } from "../../graphql/schema";

describe("StartingEquipment component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render a list of starting equipment items", () => {
    const mockStartingEquipmentData: ClassStarting_Equipment[] = [
      { equipment: { name: "Big Sword" }, quantity: 4 },
      { equipment: { name: "Leather Armor" }, quantity: 1 },
    ];

    render(
      <StartingEquipment startingEquipments={mockStartingEquipmentData} />
    );
    const startingEquipmentElement = screen.getByTestId("starting-equipment");
    const liElements = startingEquipmentElement.getElementsByTagName("li");

    expect(startingEquipmentElement).toBeInTheDocument();
    expect(liElements.length).toEqual(2);

    for (let i = 0; i < liElements.length; i += 1) {
      expect(liElements[i]).toContainHTML(
        `<li>${mockStartingEquipmentData[i].equipment!.name}: ${
          mockStartingEquipmentData[i].quantity
        }</li>`
      );
    }

    expect(startingEquipmentElement).toMatchSnapshot();
  });
});
