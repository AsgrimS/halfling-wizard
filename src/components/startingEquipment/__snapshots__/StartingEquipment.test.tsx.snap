// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartingEquipment component should render a list of starting equipment items 1`] = `
<div
  data-testid="starting-equipment"
>
  <span>
    Starting Equipment
  </span>
  <ul>
    <li>
      Big Sword
      : 
      4
    </li>
    <li>
      Leather Armor
      : 
      1
    </li>
  </ul>
</div>
`;
